lib/ts/I_Layout.h

```diff
@@ -31,80 +31,93 @@
 #ifndef _I_Layout_h
 #define _I_Layout_h
 
+#include <string>
+#include "ts/string_view.h"
+
 /**
   The Layout is a simple place holder for the distribution layout.
 
  */
 struct Layout {
-  Layout(const char *prefix = 0);
+  Layout(ts::string_view prefix = "");
   ~Layout();
 
   /**
    Return file path relative to Layout->prefix
    Memory is allocated, so use ats_free() when no longer needed
 
   */
-  char *relative(const char *file);
+  char *relative(ts::string_view file);
+
+  std::string relative_overload(ts::string_view file);
```

Comment:
This function is not named well. it should be 
`std::string relative(ts::string_view file) const;`